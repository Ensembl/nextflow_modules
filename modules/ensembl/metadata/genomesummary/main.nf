// See the NOTICE file distributed with this work for additional information
// regarding copyright ownership.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

process METADATA_GENOMESUMMARY {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "ensemblorg/ensembl-genomio:v1.6.1"

    input:
        tuple val(meta), path(input_query), path(genome_summary_json)

    output:
        tuple val(meta), path("genome.json"), emit: meta_json
        path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
        def prefix = task.ext.prefix ?: "${meta.accession}"
        def output_json = "genome.json"

        """
        genome_metadata_prepare --input_file ${input_query} --output_file ${output_json} --ncbi_meta ${genome_summary_json}

        echo -e -n "${task.process}:\n\tensembl-genomio: " > versions.yml
        genome_metadata_prepare --version >> versions.yml
        """

    stub:
        def prefix = task.ext.prefix ?: "${meta.accession}"
        def testout_json = "genome-meta.json"
        def output_json = "genome.json"

        """
        cp ${workflow.projectDir}/tests/modules/ensembl/metadata/genomesummary/${testout_json} ./${output_json}

        echo -e -n "${task.process}:\n\tensembl-genomio: " > versions.yml
        genome_metadata_prepare --version >> versions.yml
        """
}
