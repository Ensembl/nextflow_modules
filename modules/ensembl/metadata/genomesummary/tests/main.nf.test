// See the NOTICE file distributed with this work for additional information
// regarding copyright ownership.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// nf-core modules test metadata/genomesummary
nextflow_process {

    name "Test Process METADATA_GENOMESUMMARY"
    script "../main.nf"
    process "METADATA_GENOMESUMMARY"

    tag "metadata"
    tag "metadata/genomesummary"

    test("Metadata 'genome summary' test-A [real test]") {
        
        when {
            process {
                """
                input[0] = [ 
                    [accession: 'GCA_000001215.4'], 
                    file("${projectDir}/tests/modules/ensembl/metadata/genomesummary/meta-query.json", checkIfExists: true),
                    file("${projectDir}/tests/modules/ensembl/metadata/genomesummary/ncbi_genome-summary.json", checkIfExists: true)
                    ]
                """
            }
        }

        then {   
            // Since the start date and version is not constant we want to snapshot the file without them
            def genome_lines = path(process.out.meta_json.get(0).get(1)).readLines()
            def static_text = genome_lines[0..11] + genome_lines[14..19]
            assertAll(
                    { assert process.success },
                    { assert snapshot(process.out.versions).match() },
                    { assert snapshot(static_text).match("genome.json") },
                    { assert process.out.meta_json.size() == 1 },
            )
        }
    }

    test("Metadata 'genome summary' test-B [stub]") {
        
        when {
            options "-stub"
            process {
                """
                input[0] = [ 
                    [accession: 'GCA_000111222.3'], 
                    file("query.json"),
                    file("ncbi-genome.json")
                    ]
                """
            }
        }

            then {
            assertAll(
                    { assert process.success },
                    { assert snapshot(process.out).match() },
                    { assert process.out.meta_json.size() == 1 },
                    { assert process.out.versions != null }
            )
        }
    }
}
