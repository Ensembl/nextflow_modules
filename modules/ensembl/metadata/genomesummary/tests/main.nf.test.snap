{
    "Metadata 'genome summary' test-B [stub]": {
        "content": [
            {
                "0": [
                    [
                        {
                            "accession": "GCA_000111222.3"
                        },
                        "genome.json:md5,bf1aaba93d39ae7146ade4f55c53a2d8"
                    ]
                ],
                "1": [
                    "versions.yml:md5,f2b370b16189c705b60651d56a20c96c"
                ],
                "meta_json": [
                    [
                        {
                            "accession": "GCA_000111222.3"
                        },
                        "genome.json:md5,bf1aaba93d39ae7146ade4f55c53a2d8"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,f2b370b16189c705b60651d56a20c96c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-06-18T11:41:51.128599882"
    },
    "Metadata 'genome summary' test-A [real test]": {
        "content": [
            [
                "versions.yml:md5,f2b370b16189c705b60651d56a20c96c"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-06-18T11:41:40.716467645"
    },
    "genome.json": {
        "content": [
            [
                "{",
                "    \"annotation\": {",
                "        \"provider_name\": \"GenBank\",",
                "        \"provider_url\": \"https://www.ncbi.nlm.nih.gov/datasets/genome/GCA_000001215.4\"",
                "    },",
                "    \"assembly\": {",
                "        \"accession\": \"GCA_000001215.4\",",
                "        \"provider_name\": \"GenBank\",",
                "        \"provider_url\": \"https://www.ncbi.nlm.nih.gov/datasets/genome/GCA_000001215.4\",",
                "        \"version\": 4",
                "    },",
                "    \"genebuild\": {",
                "    },",
                "    \"species\": {",
                "        \"scientific_name\": \"Drosophila melanogaster\",",
                "        \"taxonomy_id\": 7227",
                "    }",
                "}"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-06-18T11:55:06.017295472"
    }
}